src/medVtkInria/vtkImageView/vtkImageViewCornerAnnotation.cxx

```diff
@@ -51,555 +50,562 @@ vtkImageViewCornerAnnotation::~vtkImageViewCornerAnnotation()
-    }
-        else
-    {
-          sprintf(text2,"%s%s",text,rpos+15);
-    }
-        tmp = text;
-        text = text2;
-        text2 = tmp;
-        rpos = strstr(text,"<image_and_max>");
-      }
-
-      rpos = strstr(text,"<slice>");
-      while (rpos)
-      {
-        *rpos = '\0';
-        if (ia && this->ShowSliceAndImage)
-    {
-          sprintf(text2,"%sSlice: %i%s",text,slice,rpos+7);
-    }
-        else
-    {
-          sprintf(text2,"%s%s",text,rpos+7);
-    }
-        tmp = text;
-        text = text2;
-        text2 = tmp;
-        rpos = strstr(text,"<slice>");
-      }
-
-      rpos = strstr(text,"<slice_and_max>");
-      while (rpos)
-      {
-        *rpos = '\0';
-        if (ia && this->ShowSliceAndImage)
-    {
-          sprintf(text2,"%sSlice: %i / %i%s",text,slice,slice_max,rpos+15);
-    }
-        else
-    {
-          sprintf(text2,"%s%s",text,rpos+15);
-    }
-        tmp = text;
-        text = text2;
-        text2 = tmp;
-        rpos = strstr(text,"<slice_and_max>");
-      }
-
-      rpos = strstr(text,"<slice_pos>");
-      while (rpos)
-      {
-        *rpos = '\0';
-        if (ia && this->ShowSliceAndImage)
-    {
-          double *dbounds = ia->GetDisplayBounds();
-          int *dext = ia->GetDisplayExtent();
-          double pos;
-          if (dext[0] == dext[1])
-      {
-            pos = dbounds[0];
-      }
-          else if (dext[2] == dext[3])
-      {
-            pos = dbounds[2];
-      }
-          else
-      {
-            pos = dbounds[4];
-      }
-          sprintf(text2,"%s%g%s",text,pos,rpos+11);
-    }
-        else
-    {
-          sprintf(text2,"%s%s",text,rpos+11);
-    }
-        tmp = text;
-        text = text2;
-        text2 = tmp;
-        rpos = strstr(text,"<slice_pos>");
-      }
-
-      rpos = strstr(text,"<window>");
-      while (rpos)
-      {
-        *rpos = '\0';
-        if (this->ImageView)
-    {
-          if (input_type_is_float)
-      {
-            sprintf(text2,"%sWindow: %g%s",text,window,rpos+8);
-      }
-          else
-      {
-            sprintf(text2,"%sWindow: %li%s",text,windowi,rpos+8);
-      }
-    }
-        else
-    {
-          sprintf(text2,"%s%s",text,rpos+8);
-    }
-        tmp = text;
-        text = text2;
-        text2 = tmp;
-        rpos = strstr(text,"<window>");
-      }
-
-      rpos = strstr(text,"<level>");
-      while (rpos)
-      {
-        *rpos = '\0';
-        if (this->ImageView)
-    {
-          if (input_type_is_float)
-      {
-            sprintf(text2,"%sLevel: %g%s",text,level,rpos+7);
-      }
-          else
-      {
-            sprintf(text2,"%sLevel: %li%s",text,leveli,rpos+7);
-      }
+            pos_x = pos[idX];
+            pos_y = pos[idY];
+            pos_z = pos[idZ];
+
+            coord_x = coord[idX];
+            coord_y = coord[idY];
+            coord_z = coord[idZ];
+
+        }
     }
-        else
+
+    if (view2d)
```

Comment:
For the fix to work even on the output of a filter `if (view2d)` as to be replaced by `if (view2d && ia)` otherwise music will crashed when using a filter (i just found this bug)